source/simulator/core.cc

```diff
@@ -2430,7 +2465,11 @@ namespace aspect
               goto start_time_iteration;
           }
 
-        postprocess ();
+        // if we use a nonlinear solver scheme and postprocess nonlinear iterations, this function is called within
+        // solve_timestep () in the individual solver schemes
+        if ((!parameters.run_postprocessors_on_nonlinear_iterations) || parameters.nonlinear_solver == NonlinearSolver::IMPES
```

Comment:
I would prefer if we would keep it consistent, and if `run_postprocessors_on_nonlinear_iterations` is true, then every solver scheme is responsible for calling postprocess. That would make the logic in the main loop simpler, and we would have less places that need to be changed every time a new solver scheme is introduced.